content/blog/prototyping/index.md

```diff
@@ -0,0 +1,732 @@
+Other languages are often written in C/C++, and it's hard to understand what's going on under the hood.
+
+When you explore a new language or domain, it's helpful to start with a prototype rather than aiming for a full-fledged production-ready solution right away.
+Otherwise, you get stuck in minor details, and due to sunk cost fallacy, you don't want to throw away the code you wrote, so you end up keeping a suboptimal design.
+
+Rust allows you to avoid these pitfalls by being very explicit about error conditions, but also by providing escape hatches in case you choose to ignore them for a while.
+
+Perhaps the most important point, however, is that eventually management will say "ship it" and you'll have to live with the code you wrote.
+In an ideal world, you'd have plenty of time to perfect the code, but in reality, deadlines are tight and you have to make compromises.
+If that's the case, a rewrite is often not an option.
+Rust allows you to have a solid foundation from the beginning.
+Even the first version is often good enough for production.
+
+Phew, with that out of the way, let's finally dive into the practical aspects of prototyping in Rust!
+
+## What Rust Prototyping Looks Like
+
+There's an overlap between prototyping and "[easy Rust](https://www.youtube.com/watch?v=33FG6O3qejM)."
+
+You allow yourself to ignore some of the best practices for production code for a while.
+The difference is that you are aware that you are prototyping.
+It's a different mode of thinking: you are exploring!
+
+Allow yourself to take some shortcuts early on.
+During this phase, it's also fine to throw away a lot of code and ideas.
+
+Python has a few good traits that we can learn from:
+
+- fast feedback loop
+- changing your mind is easy
+- it's simple to use if you ignore the edge cases
+- there is very little boilerplate
+- it's easy to experiment and refactor
+- you can write a script in a few lines
+- no compilation step 
+
+The goal is to get as close to that experience in Rust as possible while staying true to Rust's core principles.
+We want to make changes quick and painless and rapidly iterate on your design without painting ourselves into a corner.
+
+## Tips And Tricks For Prototyping In Rust
+
+### Start small
+
+It turns out you can model a surprisingly large system with just a few types and functions!
+The main idea is to defer all the unnecessary parts to later by using a "simple Rust" if you will.
+
+It's possible, but you need to switch off your inner critic who always wants to write perfect code from the beginning.
+Don't let perfect be the enemy of good.
+Rust enables you to comfortably defer perfection.
+You can make the rough edges obvious so that you can sort them out later.
+
+One of the biggest mistakes I observe is an engineer's perfectionist instinct to jump on minor details which don't have a broad enough impact to warrant the effort.
+Take a step back and focus on the big picture.
```

Comment:
Mention getting an end to end flow working first, or at least something slightly less vague

Reply:
Done.